### ISSUE TITLE
Add profile system

### ISSUE THREAD

**Issue by ryukinix** [NewayPix/hazard-memory#27]
It should make easier to lookup at runtime:

- memory consumption
- cpu consumption
- fps
- code section bottlenecks (maybe measure parts of the gameloop)

**Comment by rafaelcn** [comment]
mm, that would be nice to have but we will have to insert quite a lot of functions between the code just to track that. TheCherno has a nice implementation on this topic and I will recommend the view of those videos:

- https://www.youtube.com/watch?v=qiD39bB7DvA
- https://www.youtube.com/watch?v=xlAH4dbMVnU&t=149s

**Comment by wesleycsj** [comment]
I have only one question, is this really necessary to the first release?

**Comment by ryukinix** [comment]
It would be good, but it is not necessary. Procrastinating the profile system will make each step more obscure about if the new features are making the engine slower and less efficient.

That could be delayed, but will have costs.

**Comment by wesleycsj** [comment]
It will be delayed only once, no more than.
